assets/js/blocks/attribute-filter/block.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * External dependencies
+ */
+import {
+	useCollection,
+	useQueryStateByKey,
+	useQueryStateContext,
+} from '@woocommerce/base-hooks';
+import { useCallback, Fragment, useEffect, useState } from '@wordpress/element';
+import { find, join, sortBy } from 'lodash';
+import { ATTRIBUTES } from '@woocommerce/block-settings';
```

Comment:
Attributes is something I consider a setting that doesn't change often, and I think getting this from the server will be more performant that more API requests. It's related to the users woo environment... like post counts or other settings we pull here.

Reply:
What potential is there for the size of the attributes data package?  Is it typical for it to be a smaller number of bytes or larger?  Reason I ask is because I wonder if it could impact time to first paint of a site loading by virtue of increasing the html package for large data sets.  That's where lazy loading of attributes might have more of an impact.  Granted, if the typical number of attributes is fairly small, this would definitely be an over-optimization.

I'm fine with not addressing this right now.

Reply:
@nerrad Attributes (these are taxonomies) can vary in size and is unknown, however, woo core already caches these in a transient and they are loaded on each page anyway (to register them as taxonomies), so there is no negative to using these as block settings as it's essentially 'free' vs an API request.